technical-reference/reference-technical-articles/automatic-income-allocation.md

```diff
@@ -0,0 +1,87 @@
+
+### Core Pook - Liquidation Income
+In Core pools, incentivized tokens is transferred to the Liquidator contract in the form of vTokens. During a liquidation transaction, the Liquidator contract will try to redeem the underlying tokens from the liquidated vTokens. If the redemption process is successful, the resulting underlying tokens will be sent to the `ProtocolShareReserve` contract. However, if the redemption fails, the underlying tokens will be added to a list of pending redemptions. Subsequently, the Liquidator contract will make further attempts to redeem the pending redemptions in subsequent liquidation transactions.
+
+### Isolated Pool - Spread Income
+
+Similar to the work done to Core Pool Spread Income.
+
+### Isolated Pool - Liquidation Income
+
+In Isolated pools, liquidations are managed by the VToken contracts. During a liquidation transaction, the seized amount for the protocol is transferred to the `ProtocolShareReserve` contract.
+This ensures that 100% of the liquidation income is directly sent to the `ProtocolShareReserve` contract, without the need for additional updates.
+
+## Distribution of Collected Incomes
+
+The distribution of collected incomes is facilitated by the `ProtocolShareReserve` contract. It supports WBNB transfers but not BNB transfers. Therefore, BNB needs to be wrapped into WBNB before transferring to the `ProtocolShareReserve` contract.
+
+### `updateAssetsState` Function
+
+The `updateAssetsState` function enables the transfer of funds to the `ProtocolShareReserve` contract. After transferring funds from the origins to the `ProtocolShareReserve` contract, the `updateAssetsState` function invoked with the following parameters:
+
+- `address comptroller`: The Comptroller where the market generated the income.
+- `address asset`: The asset transferred to the `ProtocolShareReserve`.
+- `IncomeOrigin origin`: The origin type, either "spread" or "liquidation."
+
+Within the `updateAssetsState` function, the following steps are performed:
+
+- Calculate the transferred balance, taking into account the previous balance of the asset.
+- Assign the received amount to the appropriate entry (comptroller - asset - schema).
+- Track the total balance of the received asset for future distribution.
+
+### `releaseFunds` Function
+
+The `releaseFunds` function is responsible for distributing the accumulated funds to the different destinations. This function can be invoked by anyone and follows a two-step process:
+
+1. Transfer the tokens to the destination addresses.
+2. Invoke the `updateAssetsState` function in the receiver contract.
+
+## VBNBAdmin
+
+VBNBAdmin contract is the admin of the vBNB vToken. All the other vToken contracts send revenue to PSR directly whereas vBNB sends the reserves to the admin i.e., the vBNBAdmin contract. This pattern is different for vBNB because it's not a upgradable contract. 
```

Comment:
```suggestion
`VBNBAdmin` contract is the admin of the vBNB vToken. All the other vToken contracts send revenue to PSR directly whereas vBNB sends the reserves to the admin i.e., the vBNBAdmin contract. This pattern is different for vBNB because it's not a upgradable contract. 
```